services/container-entrypoint.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright (C) 2023 Olliver Schinagl <oliver@schinagl.nl>
+#
+# A beginning user should be able to docker run image bash (or sh) without
+# needing to learn about --entrypoint
+# https://github.com/docker-library/official-images#consistency
+
+set -eu
+
+bin='semonitor.sh'
+
+# run command if it is not starting with a "-" and is an executable in PATH
+if [ "${#}" -le 0 ] || \
```

Comment:
what is this wrapper intended to do?  i'd prefer to keep the stock entrypoint logic

Reply:
What is stock? This is to follow the docker-guidelines, which says to make the container behave as the user would expect.

E.g. if no arguements are passed, or if a binary from path is launched etc etc, we don't execute semonitor. So launching it with /bin/sh; we get a shell to enter the container.

Reply:
That's not what the docker guidelines recommend, because it would become a really long confluent entrypoint. e.g. take the entrypoint script, and turn it into a huge array.

The CMD being empty, means right now it will run `semonitor`without args, e.g. `--help`.

What is your concern here? I've done probably 20 of these upstream containers (from tvheadend :p, to clamav) and they all follow the same pattern.

Reply:
If I replace ENTRYPOINT in your containerfile with [ "semonitor.sh" ] I get the same effect.  eg: 
```
✗ docker run -it 281fd1590d41 -h   
usage: semonitor.py [-h] [-a] [-b BAUDRATE] [-c COMMANDS] [-d LOGFILE] [-f] [-k KEYFILE] [-m] [-n INTERFACE] [-o OUTFILE] [-p PORTS] [-r RECORD] [-s SLAVES] [-t {2,4,n}] [-u UPDATEFILE] [-v] [-x] [datasource]

Parse Solaredge data to extract inverter and optimizer telemetry

positional arguments:
  datasource     Input filename or serial port (default: stdin)
...
```

Perhaps an example of what your wrapper enables would be helpful.

> That's not what the docker guidelines recommend

[This guide](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint) says that `The best use for ENTRYPOINT is to set the image’s main command, allowing that image to be run as though it was that command, and then use CMD as the default flags.`.  If starting required more than one step then the helper script makes sense but in this case it's only ever invoking semonitor.sh -- which itself is another wrapper.